<?xml version="1.0" encoding="UTF-8"?>
<launch>
	<arg name="map_frame_id" default="map" />
	<arg name="localization_ground_truth_frame_id" default="map_ground_truth" /> 		<!-- if not empty, uses tf to retrieve ground truth (has higher priority than gazebo_ground_truth_link) -->
	<arg name="gazebo_ground_truth_link" default="guardian::base_footprint" /> 	<!-- if not empty, uses gazebo link states service to retrieve ground truth -->
	<arg name="use_roll_pitch_yaw_angles" default="false" /> 					<!-- uses quaternions -->
	<arg name="use_degrees_in_angles" default="false" /> 						<!-- uses radians -->
	<arg name="use_millimeters_in_distances" default="false" /> 				<!-- uses meters -->
	<arg name="publish_rate" default="100.0" />
	<arg name="pose_publishers_sampling_rate" default="10" />
	<arg name="tf_lookup_timeout" default="0.15" />

	<node pkg="robot_localization_tools" type="rlt_robot_localization_error_node" name="rlt_robot_localization_error" output="screen">
		<param name="gazebo_link_state_service" type="str" value="/gazebo/get_link_state" />
		<param name="gazebo_ground_truth_link" type="str" value="$(arg gazebo_ground_truth_link)" />
		<param name="invert_tf_from_map_ground_truth_frame_id" type="bool" value="false" />
		<param name="map_ground_truth_frame_id" type="str" value="$(arg localization_ground_truth_frame_id)" />
		<param name="map_frame_id" type="str" value="$(arg map_frame_id)" />
		<param name="odom_frame_id" type="str" value="odom" />
		<param name="base_link_frame_id" type="str" value="base_footprint" />
		<param name="publish_rate" type="double" value="$(arg publish_rate)" />
		<param name="tf_lookup_timeout" type="double" value="$(arg tf_lookup_timeout)" />
		<param name="pose_publishers_sampling_rate" type="int" value="$(arg pose_publishers_sampling_rate)" />
		<param name="pose_stamped_topic" type="str" value="" />
		<param name="pose_stamped_with_covariance_pose_topic" type="str" value="/initialpose" />
		<param name="pose_error_publish_topic" type="str" value="localization_error" />
		<param name="localization_poses_publisher_topic" type="str" value="localization_poses" />
		<param name="simulation_poses_publisher_topic" type="str" value="simulation_poses" />
		<param name="use_roll_pitch_yaw_angles" type="bool" value="$(arg use_roll_pitch_yaw_angles)" />
		<param name="use_degrees_in_angles" type="bool" value="$(arg use_degrees_in_angles)" />
		<param name="use_millimeters_in_distances" type="bool" value="$(arg use_millimeters_in_distances)" />
	</node>
</launch>
